### DISCUSSION TITLE
Type, that fallbacks `null | undefined` to `[]`

### DISCUSSION THREAD

**Discussion by haalogen** [mobxjs/mobx-state-tree#1868] [Q&A] [answered]
How to make a type, that fallbacks `null | undefined` to `[]` ?
Smth like `types.optional(Type, [])`, but which thinks `null` is also an empty value

**Comment by EmilTholin** [comment, marked as answer]
Hi @haalogen!

[`types.optional` has a third optional argument `optionalValues`](https://mobx-state-tree.js.org/API/#optional) that you can use to specify what values you want to override with the `defaultValue`.

**Example**
```js
import { types } from "mobx-state-tree";

const Type = types.array(types.string);

const OtherType = types.model({
  type: types.optional(Type, [], [undefined, null])
});

const otherType = OtherType.create({
  type: null
});

console.log(otherType.type); // []
```
